gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy

```diff
@@ -108,6 +108,71 @@ class PluginTest {
         }
     }
 
+    @Test
+    void pluginAddsRightRepositories_noRepositorySet() {
+        project.buildscript {
+            dependencies {
+                classpath "com.android.tools.build:gradle:${projectDependencies.get("GRADLE_BUILD_TOOLS")}"
+                classpath 'com.jakewharton.sdkmanager:gradle-plugin:0.12.0'
+            }
+        }
+
+        def manifest = project.file("src/main/AndroidManifest.xml")
+        manifest.parentFile.mkdirs()
+        manifest.text = '<manifest xmlns:android="http://schemas.android.com/apk/res/android"  package="com.realm.test"></manifest>'
+
+        project.apply plugin: 'com.android.application'
+        project.apply plugin: 'realm-android'
+
+        project.android {
+            compileSdkVersion 27
+
+            defaultConfig {
+                minSdkVersion 16
+                targetSdkVersion 27
+            }
+        }
+
+        project.evaluate()
+
+        assertTrue(project.repositories.size() == 1)
```

Comment:
This doesn't assert that a repository was added to the `project.buildscript` bloc. It actually tests for the `project.repositories`. Although plugins could, in theory, change the `project.buildscript.repositories` container, it has no effect as the `project.buildscript` DSL is only used for building the classpath for Gradle to run not the compile classpath. There is no change to this code sample required but the `pluginAddsRightRepositories_withRepositoriesSet` test will need to be fixed.